neural_compressor/compression/pruner/README.md

```diff
@@ -345,7 +363,100 @@ The following section exemplifies how to use hooks in user pass-in training func
   compression_manager.callbacks.on_train_end()
   ```
 
-In the case mentioned above, pruning process can be done by pre-defined hooks in Neural Compressor. Users need to place those hooks inside the training function.
+  In the case mentioned above, pruning process can be done by pre-defined hooks in Neural Compressor. Users need to place those hooks inside the training function.
+
+
+### Retrain-free Pruning API
+- Step 1: Define a dict-like configuration in your training codes. Usually only 5-7 configuration items need to be identified. 
+
+  If the name of the layer to be pruned is known, you can create a pruning_config manually by inputting the relevant information. This allows for greater customization and control over the pruning process.
+
+  ```python
+  pruning_configs = [
+      {  # config of a single pruner
+          "pruning_type": "retrain_free",
+          "pruning_scope": "global",
+          "op_names": [".fc", ".mlp"],  # MLP layer_names
+          "start_step": 1,
+          "end_step": 300,  # set end_step for Few shot pruning.
+          "excluded_op_names": ["lm_head"],  # A list of modules that would not be pruned.
+          "target_sparsity": 0.2,  # Target sparsity ratio of modules.
+          "pruning_frequency": 50,  # Frequency of applying pruning, The recommended setting is one fortieth of the pruning steps.
+          "pattern": "channelx1",  # Default pruning pattern.
+          "pruning_op_types": ["Linear"],
```

Comment:
currently we only support linear layer, I suggest to not expose purning_op_types  to users